graphblas/io.py

```diff
@@ -590,7 +604,21 @@ def mmread(source, *, dup_op=None, name=None):
         from scipy.sparse import isspmatrix_coo
     except ImportError:  # pragma: no cover (import)
```

Comment:
This is a minor point (and not sure if this is in line with 'best practices'), but perhaps this import check for `scipy.sparse` can be made conditional on `engine='scipy'`? (and same for fast matrix market)

Reply:
Ah, I know why I did it this way--`scipy` is needed _regardless_ of backend. I added a code comment saying as much.

But, yes, I 100% agree that we should only import libraries if they are needed.